mypy/semanal.py

```diff
@@ -1483,7 +1486,8 @@ def visit_import_all(self, i: ImportAll) -> None:
             self.add_submodules_to_parent_modules(i_id, True)
             for name, node in m.names.items():
                 node = self.normalize_type_alias(node, i)
-                if not name.startswith('_') and node.module_public:
```

Comment:
If we have access to `__all__` here, not introducing a new flag does sound better! I wasn't sure of where we had access to what, so I was somewhat following how https://github.com/python/mypy/pull/1640 implemented __all__ originally.

Thank you for mentioning this, I'll change this.